map/extrapolation/extrapolator.cpp

```diff
@@ -0,0 +1,172 @@
+#include "map/extrapolation/extrapolator.hpp"
+
+#include "geometry/distance_on_sphere.hpp"
+
+#include "platform/platform.hpp"
+
+#include "base/logging.hpp"
+
+#include <chrono>
+#include <memory>
+
+namespace
+{
+uint64_t constexpr kMaxExtrapolationTimeMs = 1000;
+uint64_t constexpr kExtrapolationPeriodMs = 200;
+double constexpr kMaxExtrapolationSpeedMPS = 120.0;
+
+double Delta(double param1, double param2, uint64_t timeBetweenPointsMs, uint64_t timeAfterParam2Ms)
+{
+  return (param2 - param1) * timeAfterParam2Ms / timeBetweenPointsMs;
+}
+
+class LinearExtrapolator
```

Comment:
BTW, what's the reason in degree 1 extrapolation? Usually it works not very well, moreover, when timeBetweenPointMs is small it's quite noisy.

Reply:
Yes. We're planning to write something for measuring for extrapolation quality and try different ways of extrapolation. Let's consider the linear one as the first and simplest version.